netdeploy/network.go

```diff
@@ -57,25 +58,33 @@ type Network struct {
 
 // CreateNetworkFromTemplate uses the specified template to deploy a new private network
 // under the specified root directory.
-func CreateNetworkFromTemplate(name, rootDir, templateFile, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
+func CreateNetworkFromTemplate(name, rootDir, templateFile string, templateReader io.Reader, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
 	n := Network{
 		rootDir:          rootDir,
 		nodeExitCallback: nodeExitCallback,
 	}
 	n.cfg.Name = name
 	n.cfg.TemplateFile = templateFile
 
-	template, err := loadTemplate(templateFile)
+	var err error
+	var template NetworkTemplate
+
+	if templateFile != "" {
+		template, err = loadTemplate(templateFile)
+	} else {
+		err = loadTemplateFromReader(templateReader, &template)
+	}
+
+	err = template.Validate()
```

Comment:
Validate needs to be called after updating the `DevMode` flag. One of the things it validates is that the dev-mode network only has a single node.

Reply:
OK